bears/general/VariableCasingBear.py

```diff
@@ -0,0 +1,143 @@
+                    diff = Diff(file)
+                    first_line = -1
+                    num_changes = 0
+                    skip_var = False
+                    for line_number, line in enumerate(file, start=1):
+                        rep = re.sub("(?P<g1>[\\" + delim + "])" + prev +
+                                     "(?P<g2>[\\" + delim + "])",
+                                     "\g<g1>" + new + "\g<g2>", line)
+                        if rep != line:
+                            if line_number in lines_changed:
+                                skip_var = True
+                            lines_changed.add(line_number)
+                            diff.change_line(line_number, line, rep)
+                            num_changes += 1
+                            if first_line == -1:
+                                first_line = line_number
+                    if skip_var:
+                        continue
+                    msg = "Change '" + prev + "' to '" + new + "'"
+                    if num_changes > 1:
+                        msg += ": " + str(num_changes) + " lines affected"
+                    result_texts.append(msg)
+                    vio = "The following name changes are suggested:"
+                    vio += "".join("\n- " + string
+                                   for string in result_texts)
+                    yield Result.from_values(
+                        self,
+                        vio,
+                        diffs={filename: diff},
+                        file=filename,
+                        line=first_line)
+                    result_texts = []
+        else:
+            self.warn("keywords and special_chars not defined for", language)
+            self.warn("Not running VariableCasingBear")
+
+    def all_to_camel(self, line):
+        i = 1
+        flast = 0
+        if line[-1:] == "_":
+            flast = 1
+        end = len(line) - flast
+        rep = line[:1].lower()
+        while i < end:
+            if line[i] == '_' and line[i + 1].islower():
+                rep += line[i + 1].upper()
+                i += 1
+            else:
+                rep += line[i]
+            i += 1
+        if flast:
+            rep += line[-1:]
+        return rep
+
+    def all_to_pascal(self, line):
+        camel = self.all_to_camel(line)
+        camel = camel[0].upper() + camel[1:]
+        return camel
+
+    def all_to_snake(self, line):
+        i = 1
+        end = len(line)
+        rep = line[:1].lower()
+        while i < end:
+            # We only want to change complete words, not abbreviations
+            # Example: abcDef becomes abc_def
+            # but abCDEf should be ab_CD_ef (better than ab_c_d_ef)
+            if line[i].isupper() and line[i - 1].islower():
+                rep += "_"
+                rep += line[i].lower()
+            else:
+                rep += line[i]
+            i += 1
+        return rep
```

Comment:
I'm using these functions too for my `FilenameBear` that formats filenames. I move this function into an own module, then you can rebase and use them from there too :)